File: Year2/CA269_ObjectOrientedProgramming/w6/PhoneNumbers.java
import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;

public class PhoneNumbers
{
    public static void main(String [] args)
    {
        System.out.println("Enter a name and number, or a name and ? to query (!! to exit)");
        
        Map<String,String> phoneNums = new HashMap<>();
        
        Scanner in = new Scanner(System.in);
        String input = in.nextLine();
        while(!input.equals("!!")){
            
            String [] info = input.split(" ");
            String name = info[0];
            String query = info[1];
            
            if(!query.equals("?")){
                phoneNums.put(name, query);
            }
            if(phoneNums.containsKey(name) && (query.equals("?"))) {
                System.out.printf("%s has number %s\n", name, phoneNums.get(name));
            }
            else if(!phoneNums.containsKey(name) && query.equals("?")){
                System.out.printf("Sorry, there is no %s\n", name);
            }
            input = in.nextLine();
        }
        System.out.println("Bye");
    }
}

File: Year2/CA268_DataStructuresAndAlgorithms/w3_SequenceSetsMaps/phone_numbers.py
d = {}

print("Enter a name and number, or a name and ? to query (!! to exit)")

n = input()
while n != "!!":
    user_input = n.strip().split()
    name, query = user_input[0], user_input[1]
    if query != "?":
        d[name] = query
    if (name in d.keys()) and (query == "?"):
        print("{} has number {}".format(name, d[name]))
    elif (name not in d.keys()) and (query == "?"):
        print("Sorry, there is no {}".format(name))
    n = input()
print("Bye")
